Build errors with Carthage and Xcode 8

I ran into this issue initially with an existing project that had gone through the migration process to Swift 2.3, but I've also been able to reproduce it with a new "Single View Application" iOS project.

Carthage version: 0.18
Moya version: 7.0.2
Xcode version: 8.0 (8A218a)

After creating a new project, I add a Cartfile with the following contents:

```
github "Moya/Moya" ~> 7.0
```

Then:

```
$ carthage bootstrap --platform iOS
*** No Cartfile.resolved found, updating dependencies
*** Fetching Moya
*** Fetching RxSwift
*** Fetching Alamofire
*** Fetching Result
*** Checking out Alamofire at "3.5.0"
*** Checking out Result at "2.1.3"
*** Checking out ReactiveCocoa at "v4.2.2"
*** Checking out RxSwift at "2.6.0"
*** Checking out Moya at "7.0.2"
*** xcodebuild output can be found in /var/folders/d1/h0cf2xs12gv7w1wx5k41q7sc0000gp/T/carthage-xcodebuild.7jZFKG.log
*** Building scheme "Alamofire iOS" in Alamofire.xcworkspace
*** Building scheme "Result-iOS" in Result.xcodeproj
*** Building scheme "ReactiveCocoa-iOS" in ReactiveCocoa.xcworkspace
--- xcodebuild: WARNING: Unable to open project file '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa.playground' in workspace '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa.xcworkspace'.
--- xcodebuild: WARNING: Unable to open project file '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa.playground' in workspace '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa.xcworkspace'.
--- xcodebuild: WARNING: Unable to open project file '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa.playground' in workspace '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa.xcworkspace'.
--- xcodebuild: WARNING: Unable to open project file '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa.playground' in workspace '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa.xcworkspace'.
*** Building scheme "RxBlocking-iOS" in Rx.xcworkspace
--- xcodebuild: WARNING: Unable to open project file '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/RxSwift/Rx.playground' in workspace '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/RxSwift/Rx.xcworkspace'.
--- xcodebuild: WARNING: Unable to open project file '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/RxSwift/Rx.playground' in workspace '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/RxSwift/Rx.xcworkspace'.
*** Building scheme "RxCocoa-iOS" in Rx.xcworkspace
--- xcodebuild: WARNING: Unable to open project file '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/RxSwift/Rx.playground' in workspace '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/RxSwift/Rx.xcworkspace'.
--- xcodebuild: WARNING: Unable to open project file '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/RxSwift/Rx.playground' in workspace '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/RxSwift/Rx.xcworkspace'.
*** Building scheme "RxSwift-iOS" in Rx.xcworkspace
--- xcodebuild: WARNING: Unable to open project file '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/RxSwift/Rx.playground' in workspace '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/RxSwift/Rx.xcworkspace'.
--- xcodebuild: WARNING: Unable to open project file '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/RxSwift/Rx.playground' in workspace '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/RxSwift/Rx.xcworkspace'.
*** Building scheme "RxTests-iOS" in Rx.xcworkspace
--- xcodebuild: WARNING: Unable to open project file '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/RxSwift/Rx.playground' in workspace '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/RxSwift/Rx.xcworkspace'.
--- xcodebuild: WARNING: Unable to open project file '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/RxSwift/Rx.playground' in workspace '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/RxSwift/Rx.xcworkspace'.
*** Building scheme "Moya iOS" in Moya.xcodeproj
** CLEAN FAILED **


The following build commands failed:
	Check dependencies
(1 failure)
** BUILD FAILED **


The following build commands failed:
	Check dependencies
(1 failure)
/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa/Swift/SignalProducer.swift:1945:69: warning: 'buffer' is deprecated: Use properties instead. 'buffer' will be removed in RAC 5.0
/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa/Swift/SignalProducer.swift:1945:69: warning: 'buffer' is deprecated: Use properties instead. 'buffer' will be removed in RAC 5.0
/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa/Swift/SignalProducer.swift:1945:69: warning: 'buffer' is deprecated: Use properties instead. 'buffer' will be removed in RAC 5.0
/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa/Swift/SignalProducer.swift:1945:69: warning: 'buffer' is deprecated: Use properties instead. 'buffer' will be removed in RAC 5.0
/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa/Swift/SignalProducer.swift:1945:69: warning: 'buffer' is deprecated: Use properties instead. 'buffer' will be removed in RAC 5.0
/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa/Swift/SignalProducer.swift:1945:69: warning: 'buffer' is deprecated: Use properties instead. 'buffer' will be removed in RAC 5.0
/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa/Swift/SignalProducer.swift:1945:69: warning: 'buffer' is deprecated: Use properties instead. 'buffer' will be removed in RAC 5.0
/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa/Swift/SignalProducer.swift:1945:69: warning: 'buffer' is deprecated: Use properties instead. 'buffer' will be removed in RAC 5.0
/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa/Swift/SignalProducer.swift:1945:69: warning: 'buffer' is deprecated: Use properties instead. 'buffer' will be removed in RAC 5.0
/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa/Swift/SignalProducer.swift:1945:69: warning: 'buffer' is deprecated: Use properties instead. 'buffer' will be removed in RAC 5.0
/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa/Swift/SignalProducer.swift:1945:69: warning: 'buffer' is deprecated: Use properties instead. 'buffer' will be removed in RAC 5.0
/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa/Swift/SignalProducer.swift:1945:69: warning: 'buffer' is deprecated: Use properties instead. 'buffer' will be removed in RAC 5.0
/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa/Swift/SignalProducer.swift:1945:69: warning: 'buffer' is deprecated: Use properties instead. 'buffer' will be removed in RAC 5.0
/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa/Swift/SignalProducer.swift:1945:69: warning: 'buffer' is deprecated: Use properties instead. 'buffer' will be removed in RAC 5.0
/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa/Swift/SignalProducer.swift:1945:69: warning: 'buffer' is deprecated: Use properties instead. 'buffer' will be removed in RAC 5.0
/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa/Swift/SignalProducer.swift:1945:69: warning: 'buffer' is deprecated: Use properties instead. 'buffer' will be removed in RAC 5.0
A shell task (/usr/bin/xcrun xcodebuild -project /Users/tim/tmp/MoyaTest/Carthage/Checkouts/Moya/Moya.xcodeproj -scheme "Moya iOS" -configuration Release -sdk iphoneos ONLY_ACTIVE_ARCH=NO BITCODE_GENERATION_MODE=bitcode CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES clean build) failed with exit code 65:
** CLEAN FAILED **


The following build commands failed:
	Check dependencies
(1 failure)
** BUILD FAILED **


The following build commands failed:
	Check dependencies
(1 failure)
```

The end of the build log shows what I think is the root issue:

```
/usr/bin/xcrun xcodebuild -project /Users/tim/tmp/MoyaTest/Carthage/Checkouts/Moya/Moya.xcodeproj -scheme "Moya iOS" -configuration Release -sdk iphoneos ONLY_ACTIVE_ARCH=NO BITCODE_GENERATION_MODE=bitcode CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES clean buildBuild settings from command line:
    BITCODE_GENERATION_MODE = bitcode
    CARTHAGE = YES
    CODE_SIGN_IDENTITY = 
    CODE_SIGNING_REQUIRED = NO
    ONLY_ACTIVE_ARCH = NO
    SDKROOT = iphoneos10.0

=== CLEAN TARGET Moya iOS OF PROJECT Moya WITH CONFIGURATION Release ===

Check dependencies
“Use Legacy Swift Language Version” (SWIFT_VERSION) is required to be configured correctly for targets which use Swift. Use the [Edit > Convert > To Current Swift Syntax…] menu to choose a Swift version or use the Build Settings editor to configure the build setting directly.
“Use Legacy Swift Language Version” (SWIFT_VERSION) is required to be configured correctly for targets which use Swift. Use the [Edit > Convert > To Current Swift Syntax…] menu to choose a Swift version or use the Build Settings editor to configure the build setting directly.

=== BUILD TARGET Moya iOS OF PROJECT Moya WITH CONFIGURATION Release ===

Check dependencies
“Use Legacy Swift Language Version” (SWIFT_VERSION) is required to be configured correctly for targets which use Swift. Use the [Edit > Convert > To Current Swift Syntax…] menu to choose a Swift version or use the Build Settings editor to configure the build setting directly.
“Use Legacy Swift Language Version” (SWIFT_VERSION) is required to be configured correctly for targets which use Swift. Use the [Edit > Convert > To Current Swift Syntax…] menu to choose a Swift version or use the Build Settings editor to configure the build setting directly.
```

--- a/Moya.xcodeproj/project.pbxproj
+++ b/Moya.xcodeproj/project.pbxproj
@@ -813,21 +813,45 @@
 				TargetAttributes = {
 					0630A3171BE1C84500AD4789 = {
 						CreatedOnToolsVersion = 7.0;
+						LastSwiftMigration = 0800;
+					};
+					0630A3201BE1C85000AD4789 = {
+						LastSwiftMigration = 0800;
+					};
+					0630A32B1BE1C85B00AD4789 = {
+						LastSwiftMigration = 0800;
 					};
 					091F0D331BDE998F009D0A49 = {
 						CreatedOnToolsVersion = 7.1;
+						LastSwiftMigration = 0800;
 					};
 					091F0D431BDE9A01009D0A49 = {
 						CreatedOnToolsVersion = 7.1;
+						LastSwiftMigration = 0800;
 					};
 					091F0D821BDE9D4D009D0A49 = {
 						CreatedOnToolsVersion = 7.1;
+						LastSwiftMigration = 0800;
 					};
 					554B0FB81BE7ED4D00F3D266 = {
 						CreatedOnToolsVersion = 7.1;
+						LastSwiftMigration = 0800;
+					};
+					554B0FC11BE7ED7C00F3D266 = {
+						LastSwiftMigration = 0800;
+					};
+					554B0FCB1BE7ED8700F3D266 = {
+						LastSwiftMigration = 0800;
 					};
 					554B10161BE7F11700F3D266 = {
 						CreatedOnToolsVersion = 7.1;
+						LastSwiftMigration = 0800;
+					};
+					554B101F1BE7F13600F3D266 = {
+						LastSwiftMigration = 0800;
+					};
+					554B10281BE7F14300F3D266 = {
+						LastSwiftMigration = 0800;
 					};
 				};
 			};
@@ -1417,6 +1441,7 @@
 				PRODUCT_NAME = Moya;
 				SDKROOT = macosx;
 				SKIP_INSTALL = YES;
+				SWIFT_VERSION = 2.3;
 			};
 			name = Debug;
 		};
@@ -1440,6 +1465,7 @@
 				PRODUCT_NAME = Moya;
 				SDKROOT = macosx;
 				SKIP_INSTALL = YES;
+				SWIFT_VERSION = 2.3;
 			};
 			name = Release;
 		};
@@ -1463,6 +1489,7 @@
 				PRODUCT_NAME = ReactiveMoya;
 				SDKROOT = macosx;
 				SKIP_INSTALL = YES;
+				SWIFT_VERSION = 2.3;
 			};
 			name = Debug;
 		};
@@ -1486,6 +1513,7 @@
 				PRODUCT_NAME = ReactiveMoya;
 				SDKROOT = macosx;
 				SKIP_INSTALL = YES;
+				SWIFT_VERSION = 2.3;
 			};
 			name = Release;
 		};
@@ -1509,6 +1537,7 @@
 				PRODUCT_NAME = RxMoya;
 				SDKROOT = macosx;
 				SKIP_INSTALL = YES;
+				SWIFT_VERSION = 2.3;
 			};
 			name = Debug;
 		};
@@ -1532,6 +1561,7 @@
 				PRODUCT_NAME = RxMoya;
 				SDKROOT = macosx;
 				SKIP_INSTALL = YES;
+				SWIFT_VERSION = 2.3;
 			};
 			name = Release;
 		};
@@ -1643,6 +1673,7 @@
 				PRODUCT_BUNDLE_IDENTIFIER = org.cocoapods.Moya;
 				PRODUCT_NAME = Moya;
 				SKIP_INSTALL = YES;
+				SWIFT_VERSION = 2.3;
 			};
 			name = Debug;
 		};
@@ -1665,6 +1696,7 @@
 				PRODUCT_BUNDLE_IDENTIFIER = org.cocoapods.Moya;
 				PRODUCT_NAME = Moya;
 				SKIP_INSTALL = YES;
+				SWIFT_VERSION = 2.3;
 			};
 			name = Release;
 		};
@@ -1687,6 +1719,7 @@
 				PRODUCT_BUNDLE_IDENTIFIER = org.cocoapods.ReactiveMoya;
 				PRODUCT_NAME = ReactiveMoya;
 				SKIP_INSTALL = YES;
+				SWIFT_VERSION = 2.3;
 			};
 			name = Debug;
 		};
@@ -1709,6 +1742,7 @@
 				PRODUCT_BUNDLE_IDENTIFIER = org.cocoapods.ReactiveMoya;
 				PRODUCT_NAME = ReactiveMoya;
 				SKIP_INSTALL = YES;
+				SWIFT_VERSION = 2.3;
 			};
 			name = Release;
 		};
@@ -1731,6 +1765,7 @@
 				PRODUCT_BUNDLE_IDENTIFIER = org.cocoapods.RxMoya;
 				PRODUCT_NAME = RxMoya;
 				SKIP_INSTALL = YES;
+				SWIFT_VERSION = 2.3;
 			};
 			name = Debug;
 		};
@@ -1753,6 +1788,7 @@
 				PRODUCT_BUNDLE_IDENTIFIER = org.cocoapods.RxMoya;
 				PRODUCT_NAME = RxMoya;
 				SKIP_INSTALL = YES;
+				SWIFT_VERSION = 2.3;
 			};
 			name = Release;
 		};
@@ -1774,6 +1810,7 @@
 				PRODUCT_NAME = Moya;
 				SDKROOT = appletvos;
 				SKIP_INSTALL = YES;
+				SWIFT_VERSION = 2.3;
 				TARGETED_DEVICE_FAMILY = 3;
 				TVOS_DEPLOYMENT_TARGET = 9.0;
 			};
@@ -1797,6 +1834,7 @@
 				PRODUCT_NAME = Moya;
 				SDKROOT = appletvos;
 				SKIP_INSTALL = YES;
+				SWIFT_VERSION = 2.3;
 				TARGETED_DEVICE_FAMILY = 3;
 				TVOS_DEPLOYMENT_TARGET = 9.0;
 			};
@@ -1820,6 +1858,7 @@
 				PRODUCT_NAME = ReactiveMoya;
 				SDKROOT = appletvos;
 				SKIP_INSTALL = YES;
+				SWIFT_VERSION = 2.3;
 				TARGETED_DEVICE_FAMILY = 3;
 				TVOS_DEPLOYMENT_TARGET = 9.0;
 			};
@@ -1843,6 +1882,7 @@
 				PRODUCT_NAME = ReactiveMoya;
 				SDKROOT = appletvos;
 				SKIP_INSTALL = YES;
+				SWIFT_VERSION = 2.3;
 				TARGETED_DEVICE_FAMILY = 3;
 				TVOS_DEPLOYMENT_TARGET = 9.0;
 			};
@@ -1866,6 +1906,7 @@
 				PRODUCT_NAME = RxMoya;
 				SDKROOT = appletvos;
 				SKIP_INSTALL = YES;
+				SWIFT_VERSION = 2.3;
 				TARGETED_DEVICE_FAMILY = 3;
 				TVOS_DEPLOYMENT_TARGET = 9.0;
 			};
@@ -1889,6 +1930,7 @@
 				PRODUCT_NAME = RxMoya;
 				SDKROOT = appletvos;
 				SKIP_INSTALL = YES;
+				SWIFT_VERSION = 2.3;
 				TARGETED_DEVICE_FAMILY = 3;
 				TVOS_DEPLOYMENT_TARGET = 9.0;
 			};
@@ -1913,6 +1955,7 @@
 				PRODUCT_NAME = Moya;
 				SDKROOT = watchos;
 				SKIP_INSTALL = YES;
+				SWIFT_VERSION = 2.3;
 				TARGETED_DEVICE_FAMILY = 4;
 				WATCHOS_DEPLOYMENT_TARGET = 2.0;
 			};
@@ -1937,6 +1980,7 @@
 				PRODUCT_NAME = Moya;
 				SDKROOT = watchos;
 				SKIP_INSTALL = YES;
+				SWIFT_VERSION = 2.3;
 				TARGETED_DEVICE_FAMILY = 4;
 				WATCHOS_DEPLOYMENT_TARGET = 2.0;
 			};
@@ -1961,6 +2005,7 @@
 				PRODUCT_NAME = ReactiveMoya;
 				SDKROOT = watchos;
 				SKIP_INSTALL = YES;
+				SWIFT_VERSION = 2.3;
 				TARGETED_DEVICE_FAMILY = 4;
 				WATCHOS_DEPLOYMENT_TARGET = 2.0;
 			};
@@ -1985,6 +2030,7 @@
 				PRODUCT_NAME = ReactiveMoya;
 				SDKROOT = watchos;
 				SKIP_INSTALL = YES;
+				SWIFT_VERSION = 2.3;
 				TARGETED_DEVICE_FAMILY = 4;
 				WATCHOS_DEPLOYMENT_TARGET = 2.0;
 			};
@@ -2009,6 +2055,7 @@
 				PRODUCT_NAME = RxMoya;
 				SDKROOT = watchos;
 				SKIP_INSTALL = YES;
+				SWIFT_VERSION = 2.3;
 				TARGETED_DEVICE_FAMILY = 4;
 				WATCHOS_DEPLOYMENT_TARGET = 2.0;
 			};
@@ -2033,6 +2080,7 @@
 				PRODUCT_NAME = RxMoya;
 				SDKROOT = watchos;
 				SKIP_INSTALL = YES;
+				SWIFT_VERSION = 2.3;
 				TARGETED_DEVICE_FAMILY = 4;
 				WATCHOS_DEPLOYMENT_TARGET = 2.0;
 			};

